test/tools/router_check/validation.proto

```diff
@@ -0,0 +1,101 @@
+syntax = "proto3";
+
+package envoy.RouterCheckToolSchema;
+
+import "envoy/api/v2/core/base.proto";
+import "validate/validate.proto";
+
+// [#protodoc-title: RouterCheckTool Validation]
+
+// The Validation Schema of the envoy router check test files.
+// The accepted input formats for the test are json and yaml.
+// The tool transparently converts json/yaml into this proto schema.
+message Validation {
+  // A collection of test cases.
+  repeated ValidationItem tests = 1;
+}
+
+// Schema for each test case.
+message ValidationItem {
+  // Name of the test case. There is no uniqueness constraint among the test case names.
+  // The name has to be non empty.
+  string test_name = 1 [(validate.rules).string.min_bytes = 1];
+
+  // The input constraints of the test case.
+  ValidationInput input = 2 [(validate.rules).message.required = true];
+
+  // The validations that need to be performed on the resultant route.
+  ValidationAssert validate = 3 [(validate.rules).message.required = true];
+}
+
+// Input values sent to the router that determine the returned route.
+// This includes the `pseudo-header <https://http2.github.io/http2-spec/#HttpRequest>`_ fields
+// defined in HTTP2.
+message ValidationInput {
+  // This pseudo-header field includes the authority portion of the target URI.
+  // Clients that generate HTTP/2 requests directly SHOULD use the :authority pseudo-header field
+  // instead of the Host header field.
+  string authority = 1 [(validate.rules).string.min_bytes = 1];
+
+  // The :path pseudo-header field includes the path and query parts of the target URI.
+  // This pseudo-header field MUST NOT be empty for http or https URIs.
+  // http or https URIs that do not contain a path component MUST include a value of '/'
+  // The exception to this rule is an OPTIONS request for an http or https URI that does not include
+  // a path component.
+  string path = 2 [(validate.rules).string.min_bytes = 1];
+
+  // This pseudo-header field includes the HTTP method.
+  // The options are GET, PUT, or POST. When not specified the field defaults to GET.
```

Comment:
Why only these three? I think this should allow any method?

Reply:
my bad. I had an impression that we only support only these, but i am not sure how i formed that opinion. Will correct this.